### DISCUSSION TITLE
Does Windows 10 support setting Terminal as the default console host?

### DISCUSSION THREAD

**Discussion by ExHix** [microsoft/terminal#13153] [Q&A] [answered]
I found `Default terminal application` setting in Windows 11 terminal, but couldn't find it in Windows 10.
I am using latest version of Windows 10. Is it because the system is not compatible to it?

**Comment by DHowett** [comment]
Correct, Windows 10 does not support this feature.

**Reply by Myrkie** [reply]
will this be supported in the future? Ive seen that it was supported in insider build 21390 but that was quite some time ago, if not is there a way to forward conhost to use the new terminal?

**Comment by seriousfreedom** [comment]
is it supported on windows 10 22h2 version

**Reply by Myrkie** [reply]
I am on 22H2 (build 19045.2251) windows 10 version and I can confirm this is still not a feature

**Reply by seriousfreedom** [reply]
Damn I think I'll start writing windows tutorials then.

Apparently you can just make shit up.

**Reply by Myrkie** [reply]
I take it someone on YouTube said it was available for windows 10?

**Reply by zadjii-msft** [reply]
![image](https://user-images.githubusercontent.com/18356694/204382385-a797f89c-8dff-4d58-bc80-aa43e4db04d3.png)

Once again, _Windows 10 does not support this feature_.

**Comment by Pawloland** [comment]
I don't know since when but it is finally possible. It works for me on :  22H2 19045.3031.
There are two ways to enable it:
Windows settings > Updates > For Developers > scroll down for Terminal, where you can select one of three: legacy windows console host, windows terminal or let windows decide.
You can also right click on legacy cmd window title bar > preferences > terminal tab > in right corner on the bottom you have drop down with the same 3 options as mentioned before.
Note that the exact names of the consecutive sub menus may differ, because I don't have English language set on my PC and I translated them.
As of WT 1.16.10261.0 there is no way of setting WT as the default terminal in OS from inside of WT settings (as it is possible on win 11).

**Comment by DHowett** [comment, marked as answer]
As @Pawloland notes, _this is finally supported on Windows 10!_ 🎉

After you install [KB5026435](https://support.microsoft.com/en-us/topic/may-23-2023-kb5026435-os-build-19045-3031-preview-2751b693-5544-4110-bc0c-feb8dd7336b3), you will be able to access the Default Terminal settings in the console host and Developer settings page.

<img width="285" alt="image" src="https://github.com/microsoft/terminal/assets/189190/f1ef0d07-be4d-4a32-b393-b23528a46d99">

<img width="608" alt="image" src="https://github.com/microsoft/terminal/assets/189190/ae24a502-cfd1-412a-a550-702ae7db2bf0">

After you install Windows Terminal v1.17, you will also be able to access the setting inside Terminal. _Default terminal is supported in earlier versions,_ but there is no code in those versions to detect that Windows 10 supports it... so it won't give you the option.

<img width="876" alt="image" src="https://github.com/microsoft/terminal/assets/189190/d908ad6f-a021-4542-a652-aa611b93e84c">

**Comment by gweinfurther** [comment]
This doesn't work. I changed this setting in both places, but Console Host still comes up as the default in Windows 10.

**Reply by zadjii-msft** [reply]
Are you running the command line application as an Administrator?

**Reply by jaredbrogan** [reply]
I'm seeing this as well. PowerShell and CMD will launch in Terminal as expected when in non-admin mode, but will not honor the setting when launching either of those as admin.

**Comment by Dragodraki** [comment]
Windows Registry Editor Version 5.00

; Use Console Host (for running as this user)
[HKEY_CURRENT_USER\Console\%%Startup]
"DelegationConsole"="{B23D10C0-E52E-411E-9D5B-C09FDF709C7D}"
"DelegationTerminal"="{B23D10C0-E52E-411E-9D5B-C09FDF709C7D}"



Windows Registry Editor Version 5.00

; Use Windows Terminal (for running as this user)
[HKEY_CURRENT_USER\Console\%%Startup]
"DelegationConsole"="{2EACA947-7F5F-4CFA-BA87-8F7FBEEFBE69}"
"DelegationTerminal"="{E12CFF52-A866-4C77-9A90-F570A7AA2C6B}"


Take note that HKCU ist relative - its a termination for what is current for the process instance. That means elevated instances with the current user not being member of administrators are different from your desktop user. Its only important if you edit the registry manually - I recommend to import this with the same script/program and the same elevation that also invokes your command prompt - that guarantees that the CMD's design match your setting no matter the user account that script/program was launched with.
